sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala

```diff
@@ -341,9 +343,14 @@ object FunctionRegistry {
   /** See usage above. */
   def expression[T <: Expression](name: String)
       (implicit tag: ClassTag[T]): (String, (ExpressionInfo, FunctionBuilder)) = {
+    expression(name, tag.runtimeClass.asInstanceOf[Class[T]])
+  }
 
+  def expression[T <: Expression](
+      name: String,
+      clazz: Class[T]): (String, (ExpressionInfo, FunctionBuilder)) = {
```

Comment:
you don't need to change these right? IIUC the only change that's necessary in this file is adding `functionExists`.